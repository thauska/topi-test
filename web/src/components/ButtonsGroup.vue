<template>
<div class="slds-col">
  <div class="slds-col slds-size_1-of-4">
    Ãšltimo Ano
  </div>
  <div class="slds-col slds-size_3-of-4">
    <ul class="slds-button-group-row">
      <li class="slds-button-group-item">
        <div class="slds-dropdown-trigger slds-dropdown-trigger_clicked">
          <button
            class="slds-button slds-button_icon slds-button_icon-more"
            aria-haspopup="true"
            title="More Options"
          >
            <svg class="slds-button__icon" aria-hidden="true">
              <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#settings" />
            </svg>
            <span class="slds-assistive-text">More options</span>
            teste
          </button>
        </div>
      </li>
      <li class="slds-button-group-item slds-m-bottom_x-small">
        <div class="slds-dropdown-trigger slds-dropdown-trigger_clicked">
          <button
            class="slds-button slds-button_icon slds-button_icon-more"
            aria-haspopup="true"
            title="More Options"
          >
            <svg class="slds-button__icon" aria-hidden="true">
              <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#table" />
            </svg>
            <span class="slds-assistive-text">More options</span>
          </button>
        </div>
      </li>
      <li class="slds-button-group-item">
        <button class="slds-button slds-button_icon slds-button_icon-border-filled">
          <svg class="slds-button__icon" aria-hidden="true">
            <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#edit" />
          </svg>
          <span class="slds-assistive-text">Provide description of action</span>
        </button>
      </li>
      <li class="slds-button-group-item">
        <button class="slds-button slds-button_icon slds-button_icon-border-filled">
          <svg class="slds-button__icon" aria-hidden="true">
            <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#refresh" />
          </svg>
          <span class="slds-assistive-text">Provide description of action</span>
        </button>
      </li>
      <li class="slds-button-group-item">
        <div class="slds-button-group" role="group">
          <button class="slds-button slds-button_icon slds-button_icon-border-filled">
            <svg class="slds-button__icon" aria-hidden="true">
              <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#chart" />
            </svg>
            <span class="slds-assistive-text">Grid</span>
            Grid
          </button>
          <button class="slds-button slds-button_icon slds-button_icon-border-filled">
            <svg class="slds-button__icon" aria-hidden="true">
              <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#filter" />
            </svg>
            <span class="slds-assistive-text">List</span>
            List
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
</template>

<script>
export default {
    name: 'ButtonsGroup'
};
</script>

<style>
</style>