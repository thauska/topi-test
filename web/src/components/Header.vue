<template>
  <div class="slds-page-header slds-p-around_medium">
    <div class="slds-page-header__row">
      <div class="slds-page-header__col-title">
        <div class="slds-media">
          <div class="slds-media__figure">
            <span class="slds-icon_container slds-icon-standard-opportunity">
              <svg class="slds-icon slds-page-header__icon" aria-hidden="true">
                <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#favorite" />
              </svg>
            </span>
          </div>
          <div class="slds-text-heading_large">Stars Repos</div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
    name: 'Header'
};
</script>

<style>
</style>