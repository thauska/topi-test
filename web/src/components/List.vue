<template>
  <div class="slds-grid slds-wrap">
    <div
      class="slds-col slds-size--1-of-1 slds-small-size--1-of-1 slds-medium-size--2-of-4"
      v-for="(repo, idx) in repositoriesData"
      :key="idx"
    >
      <div class="slds-box slds-box_x-small slds-m-around--x-small">
        <article class="slds-card">
            <div class="slds-card__body">
              <div
                class="slds-col slds-align_absolute-center"
              >
                <img
                  alt="Person name"
                  :src="repo.owner.avatar_url"
                  title="User avatar"
                  style="display: block; border-radius: 50%; width: 50%;"
                />
              </div>
            </div>
            <div class="slds-col slds-m-around_x-small">
              <div class="slds-media__body slds-m-vertical_small">
                <a :href="repo.html_url" style>
                  <h2 class="slds-card__header-title">
                    <strong>{{repo.full_name}}</strong>
                  </h2>
                </a>
                <div>
                  <ul
                    class="slds-col slds-grid slds-wrap slds-list_horizontal slds-has-dividers_left slds-has-block-links_space"
                  >
                    <li class="slds-item">Criado por</li>
                    <li class="slds-item">
                      <strong>{{repo.owner.login}}</strong>
                    </li>
                    <li class="slds-item">em: {{ formatDate(repo.created_at) }}</li>
                  </ul>
                </div>
              </div>
              <div class="slds-col slds-grid slds-wrap">
                <div>{{repo.description}}</div>
                <div class="slds-m-top_small">
                  <ul class="slds-list_horizontal">
                    <li class="slds-item slds-p-right_small">
                      <svg class="slds-button__icon" aria-hidden="true">
                        <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#favorite" />
                      </svg>
                      <span class="slds-p-left_xx-small">stars: {{repo.stargazers_count}}</span>
                    </li>
                    <li class="slds-item slds-p-right_medium">
                      <svg class="slds-button__icon" aria-hidden="true">
                        <use xlink:href="@/assets/icons/utility-sprite/svg/symbols.svg#strategy" />
                      </svg>
                      <span class="slds-p-left_xx-small">forks: {{repo.forks_count}}</span>
                    </li>
                    <li class="slds-item">
                      <small>Última atualização: {{ submittedDays(repo.pushed_at) }} dias atrás por {{ repo.owner.login }} </small>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
        </article>
      </div>
    </div>
  </div>
</template>

<script>
import RepositoriesService from '@/services/RepositoriesService'
export default {
  name: "List",
  components: {},
  data() {
    return {
      RepositoriesService: new RepositoriesService(),
      repositoriesData: [],
      page: 1,
      language: 'JavaScript'
    };
  },
  mounted () {
    this.fetchRepositories()
  },
  methods: {
    fetchRepositories() {
      this.RepositoriesService.search('stars', 'desc', 1, this.language, 10)
        .then(response => {
          this.page += 1;
          this.repositoriesData.push(...response.items)
        })
        .catch(error => { console.log(error) })
    },
    submittedDays(date) {
				const today = new Date();
				const createdDate = new Date(date);
				const days = Math.floor((today-createdDate) / (1000 * 3600 * 24));
				return  days;
    },
    formatDate(date) {
      const created = new Date(date).toLocaleDateString('pt-BR');
      return created
    }
  },
}
</script>