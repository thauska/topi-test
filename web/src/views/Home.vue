<template>
  <div class="slds-scope">
    <Header />
    <div class="myapp slds-m-horizontal_large">
      <Search />
      <List />
    </div>
  </div>
</template>

<script>
import Header from "@/components/Header";
import Search from "@/components/Search";
import List from "@/components/List";
export default {
  name: "HomePage",
  components: {
    Header,
    Search,
    List
  },
};
</script>