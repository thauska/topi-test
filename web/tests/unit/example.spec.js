import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home.vue'
import Header from '@/components/Header'
import List from '@/components/List'
import ButtonsGroup from '@/components/ButtonsGroup'

describe('Home.vue', () => {
  it('renders props.msg when passed', () => {
    const msg = 'new message'
    const wrapper = shallowMount(Home)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main child components', () => {
    // arrange
    const wrapper = shallowMount(Home)
    const listRepos = wrapper.find(List)
    const groupButtons = wrapper.find(ButtonsGroup)

    // assert
    expect(listRepos.exists()).toBe(true)
    expect(groupButtons.exists()).toBe(true)
  })
})
