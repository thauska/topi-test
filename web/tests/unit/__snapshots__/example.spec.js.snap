// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders props.msg when passed 1`] = `
<div class="slds-scope">
  <header-stub></header-stub>
  <div class="myapp slds-m-horizontal_large">
    <buttonsgroup-stub></buttonsgroup-stub>
    <list-stub></list-stub>
  </div>
</div>
`;
